shoop/configuration.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shoop.
+#
+# Copyright (c) 2012-2015, Shoop Ltd. All rights reserved.
+#
+# This source code is licensed under the AGPLv3 license found in the
+# LICENSE file in the root directory of this source tree.
+from __future__ import unicode_literals
+
+from shoop.core import cache
+from shoop.core.models import Shop, Configuration
+
+
+SHOP_CONFIGURATION_CACHE_PREFIX = "shop_configuration_%s"
+
+
+def get_cache_key(shop):
+    return SHOP_CONFIGURATION_CACHE_PREFIX % shop.pk if shop else ""
+
+
+def set(shop, key, data):
+    """
+    Set shop, key, data combination to configurations.
+    After configuration create or update bump cache version
+    or invalidate cache for all shops
+
+    :param shop: shop for configuration
+    :param key: configuration key
+    :param data: data set to configuration
+    """
+    conf, _ = Configuration.objects.get_or_create(shop=shop, key=key)
```

Comment:
This will end up saving the object twice, once with no `data` and again with `data`.

That seems kind of wasteful, and since this isn't an atomic operation, you could possibly end up with a race condition of some sort...